src/counts.jl

```diff
@@ -323,16 +323,13 @@ const BaseRadixSortSafeTypes = Union{Int8, Int16, Int32, Int64, Int128,
 radixsort_safe(::Type{T}) where {T<:BaseRadixSortSafeTypes} = true
 radixsort_safe(::Type) = false
 
-function addcounts_radixsort!(cm::Dict{T}, x::AbstractArray{T}) where T
-    # sort the x using radixsort
-    sx = sort(x, alg = RadixSort)::typeof(x)
-
+function _addcounts_radix_sort_loop!(cm, sx)
```

Comment:
Probably better keep the type signatures, it makes it easier to read the code.

It would also be useful to have a comment to explain that the separate function is needed because `sort` isn't inferred (is it still on 0.7?).

Reply:
Done. It looks like `sort` is inferred in Julia 0.7.